### ISSUE TITLE
api/positions not working after upgrade from 3.7 to 3.8

### ISSUE THREAD

**Issue by str666** [tananaev/traccar#2601]
api/positions stopped working after upgrade. Chrome is giving ERR_INVALID_RESPONSE and CURL raw response is
```
Device access denied - SecurityException (PermissionsManager:182 < PositionResource:78 < ...).
```
All devices are visible in web interface and nothing changed since upgrade so should not be permission issue. Any hints how to get this resolved? Thank you!

**Comment by tananaev** [comment]
Just tested and everything works correctly for me.

**Comment by sunhoww** [comment]
I am getting a few weird responses. One of them being this one.

It seems to happened when I use HTTP basic auth on _GET /api/positions_

- without params, I get the above response
- with params - `deviceId`, `from` and `to`, it responds with a GPX xml though.

There seem to be no issues with session cookies though. Other endpoints (at least `devices`, `groups`, `geofences`) seem fine with basic auth as well.

I am running a build with the latest commit 6dc245d.

**Comment by tananaev** [comment]
If you want JSON format, you have to include following header:
```
Accept: application/json
```

**Comment by tananaev** [comment]
As for "access denied" error, try to restart the server. It seems like you have some problem with permissions.

**Comment by sunhoww** [comment]
I made a clean install of the v3.8 release and replaced relevant files from the 6dc245d build.

Miraculously, everything is working as expected.

On hindsight, I had made the upgrade from v3.7 to v3.8 a dirty one. Maybe, the error was because I had inadvertently overwrote or removed something then.

**Comment by str666** [comment]
Access denied error did not disappear on first restart but now after reinstall and few service shutdowns api/positions seemd to return and all seems good at the moment, I'll observe how it goes from now on. Thank you for this great application and your help.

**Comment by kirantpatil** [comment]
I am also having the same issue on Version 3.9.

I can access devices but not positions.

Using Postman to test the api.

![screenshot from 2017-01-06 18 15 03](https://cloud.githubusercontent.com/assets/1178055/21719238/a393d210-d442-11e6-8d8a-3733d02febb7.png)

![screenshot from 2017-01-06 19 02 55](https://cloud.githubusercontent.com/assets/1178055/21719266/c5b189a0-d442-11e6-9afe-f1c2ddb86ad4.png)

**Comment by tananaev** [comment]
@kirantpatil, you have to provide `Accept` header.

**Comment by kirantpatil** [comment]
Thank you Sir, it works now.

**Comment by str666** [comment]
Restart will fix the problem for a while but later again same issue returned: .../api/positions gives **SecurityException (PermissionsManager:182 < PositionResource:65 < ...)**

Can confirm that adding this header to client side request fixed the problem:
> Accept: application/json

**Comment by tananaev** [comment]
Well, the error means that your user doesn't have access to the device.

**Comment by str666** [comment]
Yes it indicates some kind of permission issue, but **.../api/devices** works fine with exactly the same settings + privileges and **.../api/positions** works after either a) by adding this Accept header to request or b) by restarting the Traccar service.

This occurred in two different servers so it should not be because of the environment. We have customized all ports from 50xx to 60xx though but I don't think that affects. I tried to inspect wrapper.log.xxx for details but nothing new has appeared in a few days. This issue appeared only after switcing to latest Traccar version.

Same privileges work fine for web interface but I can't get **.../api/positions** working with Chrome without restarting Traccar service. For PHP CURL the _Accept_ -header fixed the issue without restart so from our application's perspective everything is working fine for now.
